pkg/ipcutil/ipcutil.go

```diff
@@ -0,0 +1,250 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ipcutil
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+	"github.com/containerd/nerdctl/v2/pkg/idutil/containerwalker"
+	"github.com/containerd/nerdctl/v2/pkg/labels"
+	"github.com/docker/go-units"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type IPCMode string
+
+type IPC struct {
+	Mode IPCMode `json:"mode,omitempty"`
+	// VictimContainer is only used when mode is container
+	VictimContainerID *string `json:"victim_container_id,omitempty"`
+
+	// HostShmPath is only used when mode is shareable
+	HostShmPath *string `json:"host_shm_path,omitempty"`
+
+	// ShmSize is only used when mode is private or shareable
+	// Devshm size in bytes
+	ShmSize string `json:"shm_size,omitempty"`
```

Comment:
Where is this IPC marshalized as JSON?
We do not use `this_json_style` in other places.

Reply:
I fixed it. I'm sorry.